qiskit/providers/backend_compat.py

```diff
@@ -153,32 +169,26 @@ def convert_to_target(
 
         in_data["qubit_properties"] = qubit_properties
 
-        if filter_faulty:
-            faulty_qubits = properties.faulty_qubits()
-
-        for name in prop_name_map.keys():
-            for qubits, params in properties.gate_property(name).items():
-                in_param = {
-                    "error": params["gate_error"][0] if "gate_error" in params else None,
-                    "duration": params["gate_length"][0] if "gate_length" in params else None,
-                }
-                inst_prop = InstructionProperties(**in_param)
-
-                if filter_faulty and (
-                    (not properties.is_gate_operational(name, qubits))
-                    or any(not properties.is_qubit_operational(qubit) for qubit in qubits)
-                ):
-                    faulty_ops.add((name, qubits))
-                    try:
-                        del prop_name_map[name][qubits]
-                    except KeyError:
-                        pass
-                    continue
-
-                if prop_name_map[name] is None:
-                    prop_name_map[name] = {}
-
-                prop_name_map[name][qubits] = inst_prop
+        for name in all_instructions:
+            try:
+                for qubits, params in properties.gate_property(name).items():
+                    if filter_faulty and (
+                        set.intersection(faulty_qubits, qubits)
+                        or not properties.is_gate_operational(name, qubits)
+                    ):
+                        faulty_ops.add((name, qubits))
+                        continue
+                    if prop_name_map[name] is None:
+                        # This instruction is tied to particular qubits
+                        # i.e. gate config is not provided, and instruction has been globally defined.
+                        prop_name_map[name] = {}
+                    prop_name_map[name][qubits] = InstructionProperties(
```

Comment:
Are you worried about a potential conflict here between the `BackendProperties` and the gate config above? I could rant about the poor interface for the v1 model objects, but if there is a mismatch here you'll potentially end up with a properties dict like:

```
{
  (0,): None,
  (1,): InstructionProperties(...),
  (3,): InstructionProperties(...),
}
```

Which I guess would work fine but is a bit weird to say the gate is an ideal on qubit 0 and has properties on 1 and 3. TBH, I don't remember how the converter handled inconsistent data in a v1 backend before so this is probably fine.

Reply:
I think providing correct gate config is responsibility of the provider. I just added logger message.